test/spec/openapi/schema.js

```diff
@@ -911,3 +911,117 @@ test('support query serialization params', async t => {
   t.equal(api.paths['/'].get.parameters[0].explode, false)
   t.equal(api.paths['/'].get.parameters[0].allowReserved, true)
 })
+
+test('add default properties for url params when missing schema', async t => {
+  const opt = {}
+
+  const fastify = Fastify()
+  await fastify.register(fastifySwagger, {
+    openapi: true
+  })
+  fastify.get('/:userId', opt, () => { })
+  await fastify.ready()
+
+  const swaggerObject = fastify.swagger()
+  const api = await Swagger.validate(swaggerObject)
+
+  const definedPath = api.paths['/{userId}'].get
+
+  t.same(definedPath.parameters[0], {
```

Comment:
Maybe change from `same` to `strictSame`

`same` checks if the object contains keys and values as expected, and if there are more keys it ignores them. 

`strictSame` checks if the object contains only the key and values of the expected object, and throws if there are additional attributes.